packages/utils/src/index.ts

```diff
@@ -10,8 +10,8 @@ export function getRegisteredStyles(
 
   classNames.split(' ').forEach(className => {
     if (registered[className] !== undefined) {
-      registeredStyles.push(`${registered[className]};`)
-    } else {
+      registeredStyles.push(registered[className] as string)
+    } else if (className) {
```

Comment:
`classNames` could have multiple sequential spaces, in which case `.split(' ')` would produce empty strings.

Reply:
a test case probably could be written for this, does it create a problem though? a `rawClassName` has already been allocated, can't this use some kind of string ropes or something to stay optimized?

Reply:
Yeah someone might have [inserted additional whitespaces](https://github.com/mui/material-ui/pull/43363) in their classnames.

So sure engines will usually use a concatenated string like I describe [here](https://romgrk.com/posts/optimizing-javascript#8-use-strings-carefully) to optimize concat operations, but adding a single `' '` is likely too small to be worth it, engines are most likely just allocating a new string for those bytes.

I wrote a [benchmark](https://github.com/romgrk/js-benchmark/blob/master/test-get-registered-styles.js) with only this difference, it does seem to have a positive impact:

![image](https://github.com/user-attachments/assets/56e931ca-8a04-4121-b20b-664d4afe6b9c)